### ISSUE TITLE
Possible missing coma in arrays

### ISSUE THREAD

**Issue by mcarton** [Manishearth/rust-clippy#1177]
We could warns on stuff like that:

``` rust
let a = &[
    -1, -2, -3 // <= no coma here
    -4, -5, -6
];
```

Should be similar to `suspicious_else_formatting`.

See also http://www.viva64.com/en/d/0430/. Cc #1175.

**Comment by bood** [comment]
I'm new here and want to start with this one. Is there a exist lint module I can start with or I need to create a new one? Please advise.

**Comment by oli-obk** [comment]
@bood: It fits well into the https://github.com/Manishearth/rust-clippy/blob/master/clippy_lints/src/formatting.rs module.

**Comment by bood** [comment]
@oli-obk Thanks. It also seems that the latest code on master fails to build with the latest rust nightly version
```
error: no field `map` on type `rustc::ty::TyCtxt<'a, 'tcx, 'tcx>`
  --> clippy_lints/src/vec.rs:51:75
   |
48 |           if_let_chain!{[
   |  _________- starting here...
49 | |             let Some((_, arg, _)) = higher::for_loop(expr),
50 | |             let Some(vec_args) = higher::vec_macro(cx, arg),
51 | |             is_copy(cx, vec_type(cx.tables.expr_ty_adjusted(arg)), cx.tcx.map.get_parent(expr.id)),
   | |                                                                           ^^^ unknown field
52 | |         ], {
53 | |             // report the error around the `vec!` not inside `<std macros>:`
54 | |             let span = cx.sess().codemap().source_callsite(arg.span);
55 | |             check_vec_macro(cx, &vec_args, span);
56 | |         }}
   | |__________- ...ending here: in this macro invocation

error: aborting due to 119 previous errors

error: Could not compile `clippy_lints`.
Build failed, waiting for other jobs to finish...
error: build failed
Boods-MacBook-Pro:rust-clippy Bood$ rustc -V
rustc 1.16.0-nightly (24055d0f2 2017-01-31)
```

**Comment by bood** [comment]
Also, I don't see a way I can detect the line break in this case. How about just checking the spaces between lhs and the binop (say space >=2 )? This of course would treat
```
let a = &[
    -1, -2, -3   -4, -5, -6
];
```
as a warning too.
Any suggestion?

**Comment by bood** [comment]
@mcarton this can be closed now

**Comment by oli-obk** [comment]
thanks for implementing this @bood !

**Comment by mcarton** [comment]
@bood And thanks for helping us keep the repo clean! FYI, you can [write “Fix #issue_id” in a PR](https://help.github.com/articles/closing-issues-via-commit-messages/) for GitHub to close it automatically on merge (which is much more reliable than me 😇 ).
